warehouse/slave/worker.go

```diff
@@ -341,6 +345,19 @@ func (w *worker) processSingleStagingFile(
 
 		eventLoader := w.encodingFactory.NewEventLoader(writer, job.LoadFileType, job.DestinationType)
 
+		// Duplicate detection by messageId
+		messageIdVal, ok := columnData["messageId"]
```

Comment:
It should be the `id` column, not the `messageId`. Also for snowflake it will be in upper case `ID`.